resources/js/Components/Table.vue

```diff
@@ -0,0 +1,229 @@
+ * order to facilitate the comparative analysis of data.
+ *
+ * The WiKit table component provides a wrapper around the common HTML table
+ * elements such as `<thead>`, `<tbody>`, `<tr>`, `<th>` and `<td>`, to apply
+ * design system styles to tabular data.
+ *
+ * Adding a `data-header` attribute to the cells allows us to maintain the column
+ * headers and display them in the table's linearized form to provide additional
+ * context.
+ *
+ * **Example:**
+ *
+ * ```html
+ * <td data-header="Column Header">Content Here</td>
+ * ```
+ */
+	const Table = ref('Table');
+	const props = defineProps({
+		linearize: {
+			type: String as PropType<Breakpoint>,
+			default: Breakpoint.Tablet,
+			validator: (value: Breakpoint): boolean => {
+				return validateBreakpoint( value );
+			}
+		}
+	});
+
+	const breakpoint = computed<string>(() => {
+		return validateBreakpoint( props.linearize ) ? props.linearize : 'tablet';
+	});
+
+	defineExpose({Table});
+
+</script>
+
+<style lang="scss">
+@import "@wikimedia/codex-design-tokens/theme-wikimedia-ui";
+
+	@mixin linear-table {
+		/**
+		* Completely removes thead, modern screen readers will expose the
+		* generated content
+		*/
+		thead {
+			display: none;
+			visibility: hidden;
+		}
+
+		/**
+		* Make everything display flex for alignment
+		*/
+		tbody,
+		tr,
+		th,
+		td {
+			height: auto;
+			display: flex;
+			flex-direction: column;
+		}
+
+		td,
+		th {
+			flex-direction: row;
+			flex-basis: 60%;
+		}
+
+		/**
+		* Labeling
+		*
+		* Adding a data-header attribute to the cells
+		* lets us add text before the content to provide
+		* the missing context.
+		*
+		* Markup:
+		*   <td data-header="Column Header">Content Here</td>
+		*/
+		/* stylelint-disable selector-no-qualifying-type  */
+		th[data-header]::before,
+		td[data-header]::before {
+			content: attr(data-header);
+			display: block;
+			font-weight: 700;
+			flex-basis: 40%;
+
+			// Ensure headers stay exactly 40%
+			// even if values are wider than 60%
+			min-width: 40%;
+		}
+
+		th:not([data-header]) {
+			font-weight: 700;
+		}
+
+		// Hide empty cells
+		td:empty {
+			display: none;
+		}
+	}
+
+	.Table {
+		/**
+		* Layout
+		*/
+		// As the specificationn state that the table columns should gain their
+		// width from cell content (instead of just header content / width) we
+		// revert to using the default table layout algorithm. This is done in
+		// order to avoid changing the table's display proterty and thus
+		// oblitirating it's inehrit accesibility:
+		// See: https://www.tpgi.com/short-note-on-what-css-display-properties-do-to-table-semantics/
+		table-layout: auto;
+		width: 100%;
+
+		/**
+		* Borders
+		*/
+		border-collapse: collapse;
+
+		/**
+		* Colors
+		*/
+		background-color: #fff;
+		color: #202122;
+
+		/**
+		* Typography
+		*/
+		font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Lato,Helvetica,Arial,sans-serif;
```

Comment:
Isn't this supossed to come from some of the variables or tokens that we have recently defined?